doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting *drkey.level1_db* and *drkey.secret_value_db*.
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+This could look like this::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
+
+Configuring FABRID Policies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+FABRID policies are configured in the control service using YAML files. A YAML configuration contains the information necessary to create entries in the SupportedIndicesMap, IndexIdentifierMap, IdentifierDescriptionMap (for local policies) and the MPLSMaps. Concretely the YAML file should contain the following entries:
+
+* ``local`` (bool): Indicating whether the policy is a local policy (true) or global policy (false).
+* ``local_description`` (string): The description that is fetched by remote AS'es for this specific policy. Only used for local policies, as for global policies this is stored in a global datastore. Required when ``local`` is true, ignored otherwise.
+* ``local_identifier`` (integer): The integer identifier that this policy is known by. Required when ``local`` is true, ignored otherwise.
+* ``global_identifier`` (integer): The integer identifier of the policy from the global datastore that this configured policy should implement. Required when ``local`` is false, ignored otherwise.
+* ``connections`` (list of ConnectionPoints): The connection points on which this policy applies.
+
+
+Connection Points
+'''''''''''''''''''
+A connection point in the YAML file is defined by providing the ingress and egress of the connection as well as the integer MPLS label that should be applied to enforce the policy on that connection. The egress can be either an interface, an IPv4/IPv6 prefix or a wildcard. The ingress of a connection point is limited to a wildcard or interface.The rationale behind this is that an IPv4/IPv6 ingress would indicate a packet coming from an endhost. Upon arrival at the border router, the packet would have already traversed the internal network and the border router would not be able to enforce a policy (e.g. by applying an MPLS label).
```

Comment:
- is defined by the ingress and egress... (without "by providing")
- an IPv4/IPv6, or (comma)
- interface. The (space)
- router would not have been able to enforce a policy

Reply:
Done